src/libexpr/primops.cc

```diff
@@ -112,7 +112,7 @@ StringMap EvalState::realiseContext(const NixStringContext & context)
     for (auto & outputPath : outputsToCopyAndAllow) {
         /* Add the output of this derivations to the allowed
            paths. */
-        allowPath(store->toRealPath(outputPath));
+        allowPath(outputPath);
```

Comment:
This selects a different overload of `allowPath`, which performs the `toRealPath` call, so the behavior is equivalent, and this is an improvement because it does not duplicate the logic.